src/ucs/sys/sock.c

```diff
@@ -230,6 +230,82 @@ int ucs_socket_is_connected(int fd)
     return 1;
 }
 
+
+UCS_F_NOOPTIMIZE ucs_status_t   /* GCC failed to compile it in release mode */
+ucs_socket_server_init(const struct sockaddr *saddr, socklen_t socklen,
+                       int random_port, int backlog, int *listen_fd)
+{
+    char ip_port_str[UCS_SOCKADDR_STRING_LEN];
+    struct sockaddr_in *bind_addr;
+    ucs_status_t status;
+    int ret, fd;
+
+    /* Create the server socket for accepting incoming connections */
+    status = ucs_socket_create(saddr->sa_family, SOCK_STREAM, &fd);
+    if (status != UCS_OK) {
+        goto err;
+    }
+
+    /* Set the fd to non-blocking mode (so that accept() won't be blocking) */
+    status = ucs_sys_fcntl_modfl(fd, O_NONBLOCK, 0);
+    if (status != UCS_OK) {
+        goto err_close_socket;
+    }
+
+    if (random_port) {
+        /* Loop until unused port found */
+        bind_addr = (struct sockaddr_in*)(saddr);
+        do {
+            /* Bind socket to random available port */
+            (*bind_addr).sin_port = 0;
+            ret = bind(fd, (struct sockaddr*)bind_addr, sizeof(*bind_addr));
+        } while ((ret < 0) && (errno == EADDRINUSE));
+
+        if (ret < 0) {
+            ucs_error("bind(fd=%d addr=%s) failed: %m", fd,
+                      inet_ntoa((*bind_addr).sin_addr));
+            status = UCS_ERR_IO_ERROR;
+            goto err_close_socket;
+        }
+
+        /* Get the port which was selected for the socket */
+        ret = getsockname(fd, (struct sockaddr*)bind_addr, &socklen);
+        if (ret < 0) {
+            ucs_error("getsockname(fd=%d) failed: %m", fd);
+            status = UCS_ERR_IO_ERROR;
+            goto err_close_socket;
+        }
+    } else {
+        if (bind(fd, saddr, socklen) < 0) {
+            ucs_error("bind(fd=%d addr=%s) failed: %m",
+                      fd, ucs_sockaddr_str(saddr, ip_port_str,
+                                           UCS_SOCKADDR_STRING_LEN));
```

Comment:
use sizeof(ip_port_str) in all places

Reply:
in all places in this function? file? ucx?

Reply:
where ip_port_str is used by this function

Reply:
ok, so if need to change this to:
ucs_sockaddr_str(saddr, ip_port_str, use sizeof(ip_port_str))
instead of: 
ucs_sockaddr_str(saddr, ip_port_str, UCS_SOCKADDR_STRING_LEN)
then maybe it can be in another PR? this is how it is used everywhere in ucx.

Reply:
this is a new function, so same PR

Reply:
right but ucs_sockaddr_str is used in the same way everywhere else. 
did you mean to use it differently only in this function?

Reply:
yes, see https://github.com/openucx/ucx/pull/4551#discussion_r358132951

Reply:
i wanted to have the same usage everywhere, but ok, will change only here.